cdap-common/src/main/java/io/cdap/cdap/common/http/AuthenticationChannelHandler.java

```diff
@@ -37,10 +37,16 @@
 public class AuthenticationChannelHandler extends ChannelInboundHandlerAdapter {
   private static final Logger LOG = LoggerFactory.getLogger(AuthenticationChannelHandler.class);
 
+  private final boolean enforceAuthenticatedRequests;
+
   private String currentUserId;
-  private String currentUserCredential;
+  private Credential currentUserCredential;
```

Comment:
I am wondering if userId should go inside Credential and be decoded every time it's needed. Otherwise someone can pass proper Credential, but fake UserId header

Reply:
Or at least we should validate it somehow

Reply:
It would be the same thing as using a Principal in the security context, but, if you're referring to the internal access token specifically, that already includes the username inside of the payload. Ideally, the correct way to do this would be to wrap all requests (internal or external) in the authenticated access token, otherwise a user will always be able to set the username to arbitrary values.

It should be relatively abuse-proof though since only the credential is used; I'll go ahead and add a quick doc note to the Principal class to indicate that the username may be spoofed.